# HG changeset patch
# User Rik <rik@octave.org>
# Date 1546539496 28800
#      Thu Jan 03 10:18:16 2019 -0800
# Branch stable
# Node ID bb8d66018bd7650002afb88e49abe258f56137d4
# Parent  8716297db54580afd88cfaba5d468d04544d8d3c
pt-mat.cc: Fix static analyzer detected issues (bug #55347).

* pt-mat.cc (get_concat_class): Remove useless check for "c2 == c1" done
earlier.

diff --git a/libinterp/parse-tree/pt-mat.cc b/libinterp/parse-tree/pt-mat.cc
--- a/libinterp/parse-tree/pt-mat.cc
+++ b/libinterp/parse-tree/pt-mat.cc
@@ -80,19 +80,17 @@ namespace octave
              || c1_is_logical);
 
         bool c2_is_built_in_type
           = (c2_is_int || c2_is_char ||  c2_is_double || c2_is_single
              || c2_is_logical);
 
         // Order is important here...
 
-        if (c1 == "struct" && c2 == c1)
-          retval = c1;
-        else if (c1 == "cell" || c2 == "cell")
+        if (c1 == "cell" || c2 == "cell")
           retval = "cell";
         else if (c1_is_char && c2_is_built_in_type)
           retval = c1;
         else if (c2_is_char && c1_is_built_in_type)
           retval = c2;
         else if (c1_is_int && c2_is_built_in_type)
           retval = c1;
         else if (c2_is_int && c1_is_built_in_type)
